packages/browser/src/Session.ts

```diff
@@ -187,7 +193,16 @@ export class Session extends EventEmitter {
       };
     }
     // Listen for messages from children iframes.
-    setupIframeListener(this.clientAuthentication.handleIncomingRedirect);
+    setupIframeListener(async (redirectUrl: string) => {
+      const sessionInfo = await this.clientAuthentication.handleIncomingRedirect(
+        redirectUrl
+      );
+
+      if (!isLoggedIn(sessionInfo)) {
```

Comment:
If it's possible for this handler to be invoked when already logged in, then should we still call `setSessionInfo()` regardless, i.e., maybe we were called because we want to reset our session's access token timer to a new expiration date?
Or to state this another way, is there a very specific reason that we don't want to call `setSessionInfo()`? (If so, then a comment is warranted perhaps?)

Reply:
That's a good question. @NSeydoux, do you know the answer? 'cause I don't...

Reply:
Well technically, the redirection failing does not mean that the session isn't logged out. Imagine that a user initially logs in an gets an access token valid for 30 minutes. If we try to perform a silent authentication in an iframe 10 minutes before the expiration, and it fails, then the session isn't logged out for another 10 minutes, at which point the callback we set initially when receiving the token fires and the session gets properly logged out, emitting a logout signal, clearing the storage and all. This is exaggerated, but the idea is that if the silent authentication is successful, we want to update the session info in particular with the new expiration date, but if it failed we don't do anything and have dedicated code properly logging out the session.

So long story short I think the current implementation is the "right" thing to do, but we may add a comment specifying that the proper logout happens in a callback set when receiving the access token.

Reply:
Makes sense, thanks! Added comment: 0d15be9a